libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -134,10 +136,17 @@ private void startForegroundNotification(NavigationNotification navigationNotifi
         startForeground(notificationId, notification);
     }
 
-    class LocalBinder extends Binder {
+    final static class LocalBinder extends Binder {
+
+        private final WeakReference<NavigationService> serviceRef;
+
+        private LocalBinder(final NavigationService service) {
+            this.serviceRef = new WeakReference<>(service);
+        }
+
         NavigationService getService() {
```

Comment:
This should be annotated with `@Nullable` now. Is there any caller that could hold the binder, while the service is already dead? I would think that's totally possible. Those cases need a nullcheck then

Reply:
Good call. I will add the annotation. 

Currently it is only used at one place inside the MapboxNavigation ServiceConnection. As far as I can tell it should not cause any trouble there since the usage of the service is check via `isServiceAvailable` each time as well.

Reply:
Thanks for updating this PR, even though it is unlikely, can you add a null check in MapboxNavigation where we call getService around these two lines:
```
    navigationService.startNavigation(this);
    isBound = true;
```

Maybe even setting isBound to false?

I think then this should be fine for merging.